lang/cs/Org.Apache.REEF.Client/Common/DriverFolderPreparationHelper.cs

```diff
@@ -112,25 +122,39 @@ internal void CreateDriverConfiguration(IJobSubmission jobSubmission, string dri
         /// <summary>
         /// Creates the driver folder structure in this given folder as the root
         /// </summary>
-        /// <param name="driverFolderPath"></param>
-        internal void CreateDefaultFolderStructure(string driverFolderPath)
+        /// <param name="jobSubmission">Job submission information</param>
+        /// <param name="driverFolderPath">Driver folder path</param>
+        internal void CreateDefaultFolderStructure(IJobSubmission jobSubmission, string driverFolderPath)
         {
             Directory.CreateDirectory(Path.Combine(driverFolderPath, _fileNames.GetReefFolderName()));
             Directory.CreateDirectory(Path.Combine(driverFolderPath, _fileNames.GetLocalFolderPath()));
             Directory.CreateDirectory(Path.Combine(driverFolderPath, _fileNames.GetGlobalFolderPath()));
-        }
 
-        /// <summary>
-        /// Adds all Assemlies to the Global folder in the Driver.
-        /// </summary>
-        private void AddAssemblies()
-        {
             var resourceHelper = new ResourceHelper(typeof(DriverFolderPreparationHelper).Assembly);
             foreach (var fileResources in clientFileResources)
             {
                 File.WriteAllBytes(resourceHelper.GetString(fileResources.Item1), resourceHelper.GetBytes(fileResources.Item2));
             }
+            var exePathLocal = Path.Combine(driverFolderPath, BridgeExe);
```

Comment:
Why are the two paths different?

I'm concerned that this might create debugging issues of the kind where it works on local, but doesn't work on YARN. Also, it might make porting to other RMs harder.